dev/puppet-server.conf.sample

```diff
@@ -115,3 +115,8 @@ authorization: {
         }
     ]
 }
+
+# general puppetserver settings
+puppetserver: {
+    environment-class-cache-enabled: true
+}
```

Comment:
I'm not a UX person, but I feel like if I were a user I would look at this setting and go "why the hell is that in `puppetserver`? why is there another section than `jruby-puppet`?"

I feel like our users most likely don't understand or care about the distinctions between our trapperkeeper services. I've always kind of looked at the `jruby-puppet` section as the `puppetserver` section, just with a weird name.

I guess from our (developer) perspective, putting this setting in the `jruby-puppet` section makes the code a little bit messier, but I would still be in favor of prioritizing the user experience here.